### ISSUE TITLE
Unable to import commonjs file with absolute path

### ISSUE THREAD

**Issue by danez** [vitest-dev/vitest#2218]
### Describe the bug

When I try to import a commonjs file with an absolute path (e.g. `/tmp/something.js`) vitest fails to resolve the relative dependencies of this just-imported file.

So for example if I have 

```js
// /tmp/function.js
module.exports = require('./function2.js')
```

```js
// /tmp/function2.js
module.exports  = 2
```

and in my test I try to do
```ts
import('/tmp/function.js')
```

I get

```
Error: Cannot find module './fixture2.js'
Require stack:
- /Users/username/Projects/tests/vitest-import-absolute/tmp/fixture.js
```

Which is also not a real path.

When I try the same thing with ESM and mjs files it works.

### Reproduction

https://github.com/danez-labs/vitest-absolute-import/

Clone and run `npm test`.

There are two tests, one that fails, although it should work, the other uses `/@fs/` as a workaround and works, but it is not a real solution.

### System Info

```shell
System:
    OS: macOS 12.6.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 329.33 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.10.0 - ~/.asdf/installs/nodejs/18.10.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.19.2 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Safari: 16.1
  npmPackages:
    vite: ^3.2.0 => 3.2.0 
    vitest: ^0.24.3 => 0.24.3
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
Cannot open reproduction: 

> The 'danez-labs/vitest-absolute-import' repository doesn't contain the 'tests/main.test.ts' path in 'main'

**Comment by danez** [comment]
> Cannot open reproduction:

Oopps, yes, I renamed the test afterwards. Here is the link to the repo: https://github.com/danez-labs/vitest-absolute-import/

**Comment by sheremet-va** [comment]
This issue is cause by the same problem as #1864

**Comment by sheremet-va** [comment]
Fixed with #1868, released on 0.24.4

**Comment by danez** [comment]
Thanks for the quick fix. I can confirm this works now.
